packages/core/src/util/RecurringTaskScheduler/RecurringTaskScheduler.ts

```diff
@@ -31,13 +31,14 @@ interface TaskParams {
 }
 
 export function registerRecurringTask({every, task, key}: TaskParams) {
-  const lastFireDate = loadState(key) ?? Date.now();
+  const firingDate = loadState(key) || Date.now() + every;
+  saveState(key, firingDate);
```

Comment:
Previously we were never storing fire date before the task was fired for the first time. Therefore it would restart to `Date.now() + every` on each app reload. User reloading their app frequently could make the task never being executed.